### PR TITLE
TOOL-915 Cache

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+	for _, location := range packageToLocation {
+		if location.Scheme != "file" && location.Scheme != "" {
+			log.Debugf("Flutter dependency cache: ignoring non-file scheme package: %s", location.Path)
+			continue
+		}
+
+		// Only care about absolute paths
+		if !filepath.IsAbs(location.Path) {
+			log.Debugf("Flutter dependency cache:: ignoring relative package: %s", location.Path)
```
Comment (godrei) [inline cache.go]: why do you ignore relative paths?

File: cache.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+			continue
+		}
+
+		sep := string(os.PathSeparator)
+		location.Path = strings.TrimSuffix(location.Path, sep)
```
Comment (godrei) [inline cache.go]: why do you need to trim the suffix 2x ?

File: cache.go (hunk 1 above)
Comment (lpusok) [reply cache.go]: Only encountered relative paths that point inside the current project, not the system cache.

File: cache.go (hunk 2 above)
Comment (lpusok) [reply cache.go]: Package path the parent of "lib"

Comment (lpusok) [reply cache.go]: fixed

File: cache.go [hunk 3]
```diff
@@ -168,7 +172,7 @@ func cacheFlutterDeps(projectDir string) error {
 
 	cachePaths, err := cacheableFlutterDepPaths(packageToLocation)
 	if err != nil {
-		return err
+		return nil
```
Comment (godrei) [inline cache.go]: are you sure?

Comment (lpusok) [reply cache.go]: mistake, fixed

**Pull request merged**